internal/configs/state_store.go

```diff
@@ -205,14 +205,21 @@ func (b *StateStore) Hash(stateStoreSchema *configschema.Block, providerSchema *
 		pVal = cty.UnknownVal(schema.ImpliedType())
 	}
 
+	var providerVersionString string
+	if stateStoreProviderVersion != nil {
```

Comment:
I'm wondering doesn't this unintentionally open the gate for future bugs? Like shouldn't we be conditioning this more specifically on builtin/reattached providers?